test/cli.js

```diff
@@ -325,465 +175,340 @@ describe('CLI', () => {
+            expect(result.errorOutput).to.equal('');
+            expect(result.code).to.equal(1);
+            expect(result.output).to.contain('2 tests complete');
+            expect(result.output).to.contain('Coverage: 0.00%');
             done();
-        });
+        }).catch(done);
     });
 
     it('defaults NODE_ENV environment variable to test', (done) => {
 
-        const cli = ChildProcess.spawn('node', [labPath, 'test/cli/environment.js']);
-        let output = '';
-
-        cli.stdout.on('data', (data) => {
+        runCli(['test/cli/environment.js']).then((result) => {
 
-            output += data;
-        });
-
-        cli.stderr.on('data', (data) => {
-
-            expect(data).to.not.exist();
-        });
-
-        cli.once('close', (code, signal) => {
-
-            expect(code).to.equal(0);
-            expect(signal).to.not.exist();
-            expect(output).to.contain('1 tests complete');
+            expect(result.errorOutput).to.equal('');
+            expect(result.code).to.equal(0);
+            expect(result.output).to.contain('1 tests complete');
             done();
-        });
+        }).catch(done);
     });
 
     it('changes the NODE_ENV based on -e param', (done) => {
 
-        const cli = ChildProcess.spawn('node', [labPath, 'test/cli/environment.js', '-e', 'lab']);
-        let output = '';
-
-        cli.stdout.on('data', (data) => {
-
-            output += data;
-        });
-
-        cli.stderr.on('data', (data) => {
-
-            expect(data).to.not.exist();
-        });
-
-        cli.once('close', (code, signal) => {
-
-            expect(code).to.equal(0);
-            expect(signal).to.not.exist();
-            expect(output).to.contain('1 tests complete');
+        runCli(['test/cli/environment.js', '-e', 'lab']).then((result) => {
+
+            expect(result.errorOutput).to.equal('');
+            expect(result.code).to.equal(0);
+            expect(result.output).to.contain('1 tests complete');
             done();
-        });
+        }).catch(done);
     });
 
-    it('runs tests with "only" method when set and reports correct test count', (done) => {
-
-        const cli = ChildProcess.spawn('node', [labPath, 'test/cli/only.js']);
-        let output = '';
-
-        cli.stdout.on('data', (data) => {
-
-            output += data;
-        });
-
-        cli.stderr.on('data', (data) => {
+    it('runs tests within a nestd "only" experiment and reports ran and skipped test count', (done) => {
 
-            expect(data).to.not.exist();
-        });
+        runCli(['test/cli_only-skip/onlyExperiment.js']).then((result) => {
 
-        cli.once('close', (code, signal) => {
-
-            expect(code).to.equal(0);
-            expect(signal).to.not.exist();
-            expect(output).to.contain('1 tests complete');
-            done();
-        });
+            expect(result.errorOutput).to.equal('');
+            expect(result.output).to.contain('Should execute before 1');
```

Comment:
Didn't know that, thx. Normally I use a [completely different assertion library](https://github.com/rluba/hamjest) because it provides far superior error messages.